pydantic/fields.py

```diff
@@ -244,7 +244,7 @@ class MyModel(pydantic.BaseModel):
         return cls(default=default, **kwargs)
 
     @classmethod
-    def from_annotation(cls, annotation: type[Any]) -> typing_extensions.Self:
+    def from_annotation(cls, annotation: type[Any]) -> FieldInfo:
```

Comment:
I think we still want to use `typing_extensions.Self` for these two functions, as we're returning `cls(...)`. Probably fine given that it seems like we're not subclassing, but I'd prefer to leave as is.

Reply:
These two being this one and the other one that you changed, below

Reply:
Other than that, things look great.

Reply:
> I think we still want to use `typing_extensions.Self` for these two functions, as we're returning `cls(...)`

The thing is we don't always return `cls(...)`, we sometimes make use of `cls.merge_field_infos` which returns `FieldInfo`. So to be correct the return type should be `Self | FieldInfo`, which doens't look that pretty. I can still use `Self` but with a type ignore in the body of this method though.

Reply:
You're right, let's stick with `FieldInfo`. I'll look into making those static methods as well down the line just for consistency and clarity's sake.